lib/puppet/application/master.rb

```diff
@@ -186,10 +186,11 @@ def main
 
     if Puppet.features.root?
       begin
-        Puppet::Util.chuser
+        if Etc.getpwnam(Puppet[:user])
+          Puppet::Util.chuser
+        end
       rescue => detail
-        Puppet.log_exception(detail, "Could not change user to #{Puppet[:user]}: #{detail}")
-        exit(39)
+        Puppet.warning "Could not change user to #{Puppet[:user]}: #{detail}. Continuing as root."
```

Comment:
Since we only `chuser` when the puppet user exists, I think we want to keep the fail behavior.